homeassistant/components/amcrest/helpers.py

```diff
@@ -1,13 +1,12 @@
 """Helpers for amcrest component."""
+from homeassistant.util import slugify
+
 from .const import DOMAIN
 
 
-def service_signal(service, ident=None):
-    """Encode service and identifier into signal."""
-    signal = f"{DOMAIN}_{service}"
-    if ident:
-        signal += f"_{ident.replace('.', '_')}"
-    return signal
+def service_signal(service, *args):
+    """Encode signal."""
+    return slugify("_".join([DOMAIN, service, *args]))
```

Comment:
```suggestion
    return "_".join([DOMAIN, service, *args])
```

`slugify` is doing this under the hood:
```
re.sub(r'[-\s]+', '-',
            unicode(
                re.sub(r'[^\w\s-]', '',
                    unicodedata.normalize('NFKD', string)
                    .encode('ascii', 'ignore'))
                .strip()
                .lower()))
```

```
>>> from homeassistant.components.amcrest.helpers import service_signal
>>> from homeassistant.components.amcrest.const import SERVICE_UPDATE, DOMAIN
>>> service_signal(SERVICE_UPDATE,"name","code")
'amcrest_update_name_code'
>>> "_".join([DOMAIN, SERVICE_UPDATE, "name", "code"])
'amcrest_update_name_code'
```

I'm missing why its needed here

Reply:
Yes, thank you, but I know what `slugify` does. I've looked at its code before.

This will give you an idea why I was using it:
```python
>>> from homeassistant.components.amcrest.const import DOMAIN, SERVICE_EVENT, SERVICE_UPDATE
>>> from homeassistant.components.amcrest.helpers import service_signal
>>> service_signal(SERVICE_UPDATE, "Living Room IPC")
'amcrest_update_living_room_ipc'
>>> "_".join([DOMAIN, SERVICE_UPDATE, "Living Room IPC"])
'amcrest_update_Living Room IPC'
>>> service_signal(SERVICE_EVENT, "Living Room IPC", "VideoMotion")
'amcrest_event_living_room_ipc_videomotion'
>>> "_".join([DOMAIN, SERVICE_UPDATE, "Living Room IPC", "VideoMotion"])
'amcrest_update_Living Room IPC_VideoMotion'
```
But as I said above, I'll remove it.